astropy/utils/tests/test_decorators.py

```diff
@@ -144,6 +165,14 @@ class TB(metaclass=TMeta):
     pass
 
 
+@deprecated('100.0', warning_type=NewDeprecationWarning)
+class TC:
```

Comment:
Is there a reason why this is needed outside the scope of `test_deprecated_class`?

p.s. `TA` and `TB` also look like a mess but cleaning them up is out of scope here.

Reply:
I agree that it wasn't clear whether some magic is going on I don't just get or it's just a bit "organic", so it just followed the case of `TA` and `TB` here.